store.py

```diff
@@ -22,65 +22,182 @@
+        elif name == '_self_serve_count':
+            return self._self_serve_count
+        elif name == '_line_capacity':
+            return self._line_capacity
+        else:
+            return 0
+
+
+    def get_line_list(self) -> List:
+        """Return the list of lines in this
+
         """
+        result = []
+        for line in self._line_list:
+            result.append(line)
+        return result
 
     def enter_line(self, customer: Customer) -> int:
         """Pick a new line for <customer> to join.
 
         Return the index of the line that the customer joined.
         Must use the algorithm from the handout.
 
+        When a new customer arrives, they join the open line with the fewest
+        number of customers that they are allowed to join, choosing the one
+        with the lowest index (as represented by the grocery store)
+        if there is a tie.
+
+        When there are no lines the customer can join, the “new customer” event
+        should go back into the container, and have its timestamp increased by
+        1 (representing trying to join a line again at the next time interval.)
+
         Return -1 if there is no line available for the customer to join.
+        >>> import io
+        >>> config_file = io.StringIO('{"regular_count":0,"express_count":0,"self_serve_count":1,"line_capacity":10}')
+        >>> g = GroceryStore(config_file)
+        >>> customer = Customer('bill nye', [Item('banana', 5), Item('apple', 6)])
+        >>> g.enter_line(customer)
+        0
         """
 
+        i = -1
+        for line in self._line_list:
+            if line.can_accept(customer):
+                line.queue.append(customer)
+                break
+            else:
+                i += 1
+
+        if i == len(self._line_list):
+            return -1
+        else:
+            return i
+
     def line_is_ready(self, line_number: int) -> bool:
         """Return True iff checkout line <line_number> is ready to start a
-        checkout.
+        checkout. Thus, line_is_ready should return True
+        if and only if there is exactly one customer in line
         """
 
+        line = self._line_list[line_number]
+        if len(line.queue) == 1:
+            return True
+        return False
+
     def start_checkout(self, line_number: int) -> int:
         """Return the time it will take to check out the next customer in
         line <line_number>
         """
+        next_customer = self._line_list[line_number][0]
+        return next_customer.get_item_time()
 
-    def complete_checkout(self, line_number: int) -> int:
-        """Return the number of customers remaining to be checked out in line
-        <line_number>
+    def complete_checkout(self, line_number: int) -> bool:
+        """Return True iff there are customers remaining to be checked out in
+        line <line_number>
         """
+        line = self._line_list[line_number]
+        return len(line.queue) >= 1
```

Comment:
why are you accessing object attiributes directly ? there got to be a method for that.

Reply:
its a public attribute: not as frowned upon and generally accepted in the grading scheme